### ISSUE TITLE
"rlang" package causing a load error?

### ISSUE THREAD

**Issue by StevePrecio** [prioritizr/prioritizr#123]
Perhaps some else has encountered this.  I'm now getting an error message when loading the prioritizR package because it seems to be getting stopped by the call to "rlang."  The script is in the attached file:

I'm running R v3.6.1, RStudio v1.2.5, and MacOSX v10.15

Any suggestions?

[Load_prioritizR_package.txt](https://github.com/prioritizr/prioritizr/files/3776508/Load_prioritizR_package.txt)

**Comment by jeffreyhanson** [comment]
Hi, that's odd. It looks like you might need to install the rlang package. Could you please do this (i.e. `install.packages("rlang")` and try loading prioritizr again?

**Comment by StevePrecio** [comment]
Thanks for helping me with this.  Yes, I tried that, but I get the attached error message, which seems to be driven by a compilation error due to a non-zero exit status:

[install_rlang.txt](https://github.com/prioritizr/prioritizr/files/3776714/install_rlang.txt)

**Comment by jeffreyhanson** [comment]
It looks like rlang was recently updated and so CRAN hasn't created a binary (pre-compiled) version of the package yet. So, you have three options: 

1. Install xcode and the software needed to install R packages from source. This can be tricky. I don't advise this unless you have a few hours to kill and are good at command line stuff. I don't use Mac OSX so I can't help you with this sorry.
2. Wait a couple of days for CRAN to create the binary version of the package and then install that using `install.packages("rlang")`.
3. Install the binary version of the previous version of the rlang package. You might be able to do this with the code: `install.packages("https://cran.r-project.org/bin/macosx/el-capitan/contrib/3.6/rlang_0.4.0.tgz")`. If you go with this method and that code doesn't work, let me know and we can try another method to install it.

**Comment by StevePrecio** [comment]
Again, thanks.  I'll pass on Option 1; "hours to kill" is not in the current workflow schedule.  (So to speak.).

I tried Option 3, and I got the following message: Warning in install.packages :
  package ‘https://cran.r-project.org/bin/macosx/el-capitan/contrib/3.6/rlang_0.4.0.tgz’ is not available (for R version 3.6.1)

So, I'll go with Option 2 and give it a couple of days.  I'll post a comment here about what happens on Tuesday.

Thanks!  At least now I know it isn't something stupid that I'm doing, which is sometimes the case.

**Comment by jeffreyhanson** [comment]
No worries - glad I could help. I'm sorry that code didn't work, maybe you could try this code instead to get the older version installed:

```
install.packages("http://cran.r-project.org/bin/macosx/el-capitan/contrib/3.6/rlang_0.4.0.tgz", repos = NULL)
```

**Comment by StevePrecio** [comment]
Damn, that worked!  Thanks!!!  And I'll still try the updated rlang in a couple of days without the text for the workaround.

**Comment by jeffreyhanson** [comment]
Awesome! I'm glad we got it working. Yeah, I'd recommend updating to the latest version to ensure compatibility with the tidyverse packages. I'll close this issue now since it looks like you've got it working - but please feel free to reopen it if you run into anymore issues.

**Comment by StevePrecio** [comment]
Jeff, you probably already know this, but just FYI in case others encounter it, Gurobi does not currently run with R v3.6.  I encountered a problem with it, posted a query on their message board, and got this back …


“Gurobi 8 only supports R up to version 3.5. (See here<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.gurobi.com%2Fde%2Fproducts%2Fgurobi-optimizer%2Fsupported-platforms%2F&data=02%7C01%7Ctrombula%40middlebury.edu%7Ce50d6619ebcb4f0cfeec08d75c3d707f%7Ca1bb0a191576421dbe93b3a7d4b6dcaa%7C1%7C1%7C637079291334942137&sdata=9w0fAa9cJmpsOkzNU3oX%2BIzqN6MCRk3emSwiw3VFo64%3D&reserved=0>.) It may be possible to work around this by manually adjusting the paths in the package, but this is not supported.
Gurobi 9 (scheduled for later this fall) will support R 3.6.”
Until Gurobi 9 is released, prioritizR users who set Gurobi as the optimizer should probably stick with R v3.5 unless they want to adjust the paths called.

Steve

**Comment by jeffreyhanson** [comment]
Hey - thank you very much for reporting this! I didn't know that Gurobi version 8 didn't work for R version 3.6.

**Comment by StevePrecio** [comment]
No worries.  I’m glad I wasn’t just clogging your inbox unnecessarily.

Cheers,

s

**Comment by StevePrecio** [comment]
Jeff, in case you haven’t seen it yet, Gurobi 9.0.0 was released on Friday.  It seems to work just fine with R 3.6.1.  Running through the prioritiR tutorial with both of those installed on Mac OSX and through RStudio 1.2.5, I did not encounter any problems.

Cheers,

Steve

From: "Trombulak, Stephen C." <trombula@middlebury.edu>
Date: Tuesday, October 29, 2019 at 9:21 AM
To: prioritizr/prioritizr <reply@reply.github.com>
Subject: Just FYI

Jeff, you probably already know this, but just FYI in case others encounter it, Gurobi does not currently run with R v3.6.  I encountered a problem with it, posted a query on their message board, and got this back …


“Gurobi 8 only supports R up to version 3.5. (See here<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.gurobi.com%2Fde%2Fproducts%2Fgurobi-optimizer%2Fsupported-platforms%2F&data=02%7C01%7Ctrombula%40middlebury.edu%7Ce50d6619ebcb4f0cfeec08d75c3d707f%7Ca1bb0a191576421dbe93b3a7d4b6dcaa%7C1%7C1%7C637079291334942137&sdata=9w0fAa9cJmpsOkzNU3oX%2BIzqN6MCRk3emSwiw3VFo64%3D&reserved=0>.) It may be possible to work around this by manually adjusting the paths in the package, but this is not supported.
Gurobi 9 (scheduled for later this fall) will support R 3.6.”
Until Gurobi 9 is released, prioritizR users who set Gurobi as the optimizer should probably stick with R v3.5 unless they want to adjust the paths called.

Steve
